File: app/src/main/java/com/karthik/splash/Components/SplashScreenComponent.java
package com.karthik.splash.Components;

import com.karthik.splash.Modules.SplashScreenModule;
import com.karthik.splash.Views.Splash;

import dagger.Subcomponent;

/**
 * Created by karthikrk on 12/11/17.
 */
@Subcomponent(modules = {SplashScreenModule.class})
public interface SplashScreenComponent {
    void inject(Splash splash);
}

File: app/src/main/java/com/karthik/splash/splashscreen/di/SplashScreenComponent.kt
package com.karthik.splash.splashscreen.di

import com.karthik.splash.splashscreen.SplashScreen
import dagger.Subcomponent

@SplashScreenScope
@Subcomponent(modules = [SplashScreenModule::class])
interface SplashScreenComponent {
    fun inject(splashscreen: SplashScreen)
}